# Bonus: Temperature Analysis I

In [1]:
import pandas as pd
from datetime import datetime as dt

In [5]:
# "tobs" is "temperature observations"
df = pd.read_csv('Resources/hawaii_measurements.csv')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19550 entries, 0 to 19549
Data columns (total 4 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   station  19550 non-null  object 
 1   date     19550 non-null  object 
 2   prcp     18103 non-null  float64
 3   tobs     19550 non-null  int64  
dtypes: float64(1), int64(1), object(2)
memory usage: 611.1+ KB


In [7]:
# Convert the date column format from string to datetime
df['date'] = pd.to_datetime(df['date'])
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19550 entries, 0 to 19549
Data columns (total 4 columns):
 #   Column   Non-Null Count  Dtype         
---  ------   --------------  -----         
 0   station  19550 non-null  object        
 1   date     19550 non-null  datetime64[ns]
 2   prcp     18103 non-null  float64       
 3   tobs     19550 non-null  int64         
dtypes: datetime64[ns](1), float64(1), int64(1), object(1)
memory usage: 611.1+ KB


In [16]:
# Set the date column as the DataFrame index
df.set_index('date', inplace=True)

In [ ]:
# Drop the date column

### Compare June and December data across all years 

In [62]:
from scipy import stats
from scipy.stats import ttest_rel

In [46]:
# Filter data for desired months
df_jun=df[(df.index.month == 6)]
df_dec=df[(df.index.month == 12)]

In [80]:
df_dec.info()

<class 'pandas.core.frame.DataFrame'>
DatetimeIndex: 1517 entries, 2010-12-01 to 2016-12-31
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   station  1517 non-null   object 
 1   prcp     1405 non-null   float64
 2   tobs     1517 non-null   int64  
dtypes: float64(1), int64(1), object(1)
memory usage: 47.4+ KB


In [58]:
# Identify the average temperature for June
df_jun['tobs'].mean()

74.94411764705882

In [59]:
# Identify the average temperature for December
df_dec['tobs'].mean()

71.04152933421226

In [72]:
# Create collections of temperature data
tobs_jun = df_jun.filter(['tobs'])
tobs_dec = df_dec.filter(['tobs'])
list_jun = df_jun['tobs'].tolist()
list_dec = df_dec['tobs'].tolist()

In [77]:
# Run un-paired t-test
stats.ttest_ind(list_jun, list_dec)

Ttest_indResult(statistic=31.60372399000329, pvalue=3.9025129038616655e-191)

### Analysis

####  Run unpaired t-test due to different number of days and number of samples in June (1,700) vs December (1,517).
#### The T-value is high measuring a relatively large diferrence to the vairation of data.
#### The p value obtained from the t-test is significant (p < 0.05) indicating that results are highly unlikely to be by chance.
#### Therefore, we conclude that the temperature in June is significantly different than temperature in December.